source/codegen/validate-linux-rt.py

```diff
@@ -0,0 +1,66 @@
+"""Script for validating if a set of changed files warrants updating the Linux RT Feed.
+
+Expects output of git diff --name-only <last-release-commit> <current-commit> 
+to be piped in so it can be read from sys.stdin
+"""
+
+import argparse
+import fnmatch
+import os
+import sys
+from typing import Set
+
+from template_helpers import load_metadata
+
+
+def _get_codegen_changes(changed_files: Set[str]) -> Set[str]:
+    return set(fnmatch.filter(changed_files, "source/codegen/*"))
+
+
+def _get_non_rt_driver_changes(driver_name: str, changed_files: Set[str]) -> Set[str]:
+    files_affecting_linux_rt = [
+        f"generated/{driver_name}/*",
+        f"examples/{driver_name}/*",
+        f"source/custom/{driver_name}*",
+    ]
+    non_rt_driver_changes = set()
+
+    for pattern in files_affecting_linux_rt:
+        non_rt_driver_changes |= set(fnmatch.filter(changed_files, pattern))
+    return non_rt_driver_changes
+
+
+def _need_linux_rt_feed_update(metadata_dir: str, changed_files: Set[str]) -> bool:
+    codegen_changes = _get_codegen_changes(changed_files)
+    remaining_changes = changed_files - codegen_changes
```

Comment:
Remaining question I thought of is if we can additionally ignore "source/tests/*" changes like we can codegen changes... I think that's reasonable and avoids trying to figure out if a test change is related to a non-rt driver or not. @astarche , thoughts on that?

Reply:
Actually, I went ahead and made this change. I think it's reasonable to discard test changes. If there's only test changes plus non-RT driver changes then it's probably still safe to say the set of changes isn't affecting RT.